packages/rich-text/src/plugins/Table/createTablePlugin.ts

```diff
@@ -66,10 +71,11 @@ export const createTablePlugin = (): RichTextPlugin =>
       editor.insertFragment = (fragments) => {
         // We need to make sure we have a new, empty and clean paragraph in order to paste tables as-is due to how Slate behaves
         // More info: https://github.com/ianstormtaylor/slate/pull/4489 and https://github.com/ianstormtaylor/slate/issues/4542
-        const fragmentHasTable = fragments.some(
-          (fragment) => (fragment as CustomElement).type === BLOCKS.TABLE
-        );
-        if (fragmentHasTable) {
+        const fragmentHasTable = fragments.some((fragment) => isTable(fragment as CustomElement));
+        const isFirstFragment =
```

Comment:
We only need to add a paragraph in case table is the first element of a list of multiple other items, e.g. `table + ul + p`, but if we have `p + ul + table`, then we don't need to add a new and empty paragraph.